lib/tokenlist.cpp

```diff
@@ -642,7 +642,8 @@ static bool iscpp11init_impl(const Token * const tok)
 
     const Token *endtok = nullptr;
     if (Token::Match(nameToken, "%name%|return|: {") &&
-        (!Token::simpleMatch(nameToken->tokAt(2), "[") || findLambdaEndScope(nameToken->tokAt(2))))
+        (!Token::simpleMatch(nameToken->tokAt(2), "[") || findLambdaEndScope(nameToken->tokAt(2))) &&
+        !Token::simpleMatch(nameToken->tokAt(-2), ") .") && !Token::Match(nameToken->tokAt(-3), ") &|&& .")) // trailing return type, where -> is replaced by .
```

Comment:
Pushed a fix for this. A bit different approach, using the fact that the for-loop which searches for `;` will always work for function bodies that have a non-void return-type (since there must be a `return ... ;` somewhere). The only cases I could think of that do _not_ have a semicolon in the body, are `void`-functions. So checking for `-> void {` is good enough